lib/Query_Processor.cpp

```diff
@@ -2779,6 +2779,15 @@ bool Query_Processor::query_parser_first_comment(Query_Processor_Output *qpo, ch
 					}
 				}
 			}
+			if (!strcasecmp(key,"min_gtid")) {
+				size_t vs = strlen(value);
+				if (vs > 0) {
+					char *buf=(char*)malloc(vs);
+					strncpy(buf, value, vs - 1);
```

Comment:
The `size_t strlen(const char* s)` function calculates the length of the string pointed to by its parameter, excluding the terminating null byte ('\0'). Therefor `buf` contains truncated value.